### DISCUSSION TITLE
创建一个空表时，创建rollup为什么不直接更新元数据，而是有BE等操作呢？设计理念是什么？

### DISCUSSION THREAD

**Discussion by aixuebo** [apache/doris#22102] [A - General / Q&A] [answered]
各位老板好，打扰了，我这边有一些使用doris在设计上的困惑，希望老板们答疑下。
1.创建一个空表时，按照dt创建partition，比如初始化的时候需要创建5年的partition。
很快可以把表创建完成，但此时增加若干个rollup的时候，创建rollup的时间大概需要20s，甚至有时候1分钟。
比较困惑我的是，此时是空表，而rollup也是空表基础上创建的，为什么会耗时这么久？

看了一些文章后，发现底层实现时，不仅更新元数据，而是与BE有相关操作，想问为什么会在设计上有这个步骤呢？为什么不直接更新元数据，当某些分区有数据导入的时候，在创建对应的分区BE数据呢？

2.分区为什么一定是连续的设计，为什么不是load一个分区，就产生一个分区，没有load导入数据的分区，在分区元数据里没有不可以吗？类似于hive那种。

**Comment by LemonLiTree** [comment]
1. What version are you testing on? I tested it myself on an empty table. The rollup will be completed immediately, and it doesn’t take too much time.
2. Automatic creation of partitions based on data values is already in progress

**Reply by aixuebo** [reply]
你好，可能是我表达的有问题，有时候我的版本创建的也会很快，我其实想问的是背后原理的考虑初衷。
1.我想了解的是，咱们设计上的考虑因素是什么，比如为什么以前的版本中，partition需要连续设计，而不是动态有数据的时候创建对应的partition呢？
2.为什么在空的表上，创建rollup的时候，要走BE操作，根据partition、备份数量，现阶段应该是与BE有关系的操作流程吧。而我想不明白的点是，在空表上创建rollup，只需要更新元数据就可以了，没有跟BE有关系的环节。

这两个逻辑困扰我好久，所以我来咱们社群提交一个QA，想咨询一下老板们在设计上为什么这么做，有什么好处。

**Reply by LemonLiTree** [reply, marked as answer]
1. just historical reasons
2. I think you can look at the code that creates the rollup.
